src/output.rs

```diff
@@ -333,6 +358,13 @@ impl DebugBuffer {
 
 impl fmt::Display for DebugBuffer {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        write_buffer(&self.buffer, f)
+        if self.buffer.len() > 8192 {
+            write!(f, "<{} bytes total>", self.buffer.len())?;
+            write_bytes(&self.buffer[..2048], f)?;
+            write!(f, "...<{} bytes omitted>...", self.buffer.len() - 4096)?;
+            write_bytes(&self.buffer[self.buffer.len() - 2048..], f)
```

Comment:
I'd feel more comfortable if we explicitly did the math on these.

```rust
let min_overflow = 8192;
...
let max_start = 2048;
let max_end = 2048;
let max_printed = max_start + max_end;
assert!(max_printed < max_overflow);

write!(f, "<{} bytes total>", self.buffer.len())?;
write_bytes(&self.buffer[..max_start], f)?;
write!(f, "...<{} bytes omitted>...", self.buffer.len() - max_printed)?;
write_bytes(&self.buffer[self.buffer.len() - max_end..], f)
```

Reply:
Done